### ISSUE TITLE
missing postcss dependency on build

### ISSUE THREAD

**Issue by ali-phipps** [vercel/next.js#24882]
### What version of Next.js are you using?

10.2.0

### What version of Node.js are you using?

v14.15.1

### What browser are you using?

Chrome 90

### What operating system are you using?

macOS 10.15.7

### How are you deploying your application?

next build and start on my local development env

### Describe the Bug

Upgrading from 10.1.1 to 10.2.0 in our package.json and running npm install. 

we get the following dependency error on running `next build` - 

```Failed to compile.
info  - Loaded env from /Users/tso987/Projects/standard-pdp-ui/.env.production
info  - Loaded env from /Users/tso987/Projects/standard-pdp-ui/.env
info  - Using webpack 5. Reason: future.webpack5 option enabled https://nextjs.org/docs/messages/webpack5
next-transpile-modules - WARNING experimental Webpack 5 support enabled
info  - Checking validity of types .next-transpile-modules - WARNING experimental Webpack 5 support enabled
info  - Checking validity of types  
info  - Creating an optimized production build  
Failed to compile.

HookWebpackError: PostCSS plugin postcss-discard-comments requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users


> Build error occurred
Error: > Build failed because of webpack errors
    at /Users/tso987/Projects/standard-pdp-ui/node_modules/next/dist/build/index.js:17:924
    at async Span.traceAsyncFn (/Users/tso987/Projects/standard-pdp-ui/node_modules/next/dist/telemetry/trace/trace.js:6:584)
npm ERR! code ELIFECYCLE
```
We aren't using any postcss config in our code.

running `npm ls postcss` shows the missing dependency - 

```
standard-pdp-ui@2.0.0 /Users/tso987/Projects/standard-pdp-ui
├─┬ @JohnLewisPartnership/jl-ui-kit@32.21.0
│ ├─┬ autoprefixer@9.8.6
│ │ └── postcss@7.0.35 
│ ├─┬ css-loader@1.0.1
│ │ ├─┬ icss-utils@2.1.0
│ │ │ └── postcss@6.0.23 
│ │ ├── postcss@6.0.23 
│ │ ├─┬ postcss-modules-extract-imports@1.2.1
│ │ │ └── postcss@6.0.23 
│ │ ├─┬ postcss-modules-local-by-default@1.2.0
│ │ │ └── postcss@6.0.23 
│ │ ├─┬ postcss-modules-scope@1.1.0
│ │ │ └── postcss@6.0.23 
│ │ └─┬ postcss-modules-values@1.3.0
│ │   └── postcss@6.0.23 
│ ├─┬ postcss-inline-svg@3.1.1
│ │ └── postcss@6.0.23 
│ └─┬ postcss-loader@3.0.0
│   └── postcss@7.0.35  deduped
└─┬ next@10.2.0
  └── UNMET PEER DEPENDENCY postcss@8.2.13 
```

I've then had to add postcss as a direct dependency, which then resolves the issue. 

### Expected Behavior

NPM should upgrade next's dependency tree to 10.2, and the app should build correctly

### To Reproduce

Our package.json looks like this - 
```{
  "name": "standard-pdp-ui",
  "description": "Standard product detail page",
  "private": true,
  "version": "v2.0.0",
  "config": {
    "blueprint": {
      "port": 4567,
      "http_probe": "/api/ready",
      "bucket": "standardpdp-assets",
      "hostedAssets": false
    }
  },
  "scripts": {
    "dev": "blueprint-scripts dev",
    "format": "blueprint-scripts format",
    "lint": "blueprint-scripts lint",
    "test": "blueprint-scripts test",
    "cypress": "blueprint-scripts cypress",
    "build": "blueprint-scripts build",
    "start": "blueprint-scripts start",
    "contract": "blueprint-scripts contract",
    "testcafe": "testcafe ${TESTCAFE_BROWSERS:-'chrome:headless'} testcafe/ --skip-js-errors",
    "testcafe-open": "testcafe ${TESTCAFE_BROWSERS:-'chrome'} testcafe/ --skip-js-errors"
  },
  "dependencies": {
    "@JohnLewisPartnership/jl-esi-components": "^6.0.2",
    "@JohnLewisPartnership/jl-ui-kit": "^32.17.0",
    "blueprint": "file:.blueprint",
    "@JohnLewisPartnership/browserslist-config-jl": "^1.4.1",
    "@JohnLewisPartnership/jl-analytics": "0.0.6",
    "@next/bundle-analyzer": "^10.0.9",
    "abort-controller": "^3.0.0",
    "debug": "^4.3.2",
    "deepmerge": "^4.2.2",
    "dotenv": "^8.2.0",
    "dotenv-expand": "^5.1.0",
    "graphql": "^15.5.0",
    "graphql-request": "^3.4.0",
    "identity-obj-proxy": "^3.0.0",
    "isomorphic-unfetch": "^3.1.0",
    "js-cookie": "^2.2.1",
    "newrelic": "^7.1.3",
    "next": "10.2.0",
    "next-compose-plugins": "^2.2.1",
    "next-transpile-modules": "^6.3.0",
    "nookies": "^2.5.2",
    "pino": "^6.11.2",
    "pino-http": "^5.5.0",
    "query-string": "^7.0.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "sass": "^1.32.8"
  },
  "devDependencies": {
    "@simplenode/vcr": "^1.3.0",
    "@commitlint/cli": "^11.0.0",
    "@commitlint/config-conventional": "^11.0.0",
    "@google/semantic-release-replace-plugin": "^1.0.2",
    "@pact-foundation/pact": "^9.15.3",
    "@semantic-release/changelog": "^5.0.1",
    "@semantic-release/commit-analyzer": "^8.0.1",
    "@semantic-release/exec": "^5.0.0",
    "@semantic-release/git": "^9.0.0",
    "@semantic-release/gitlab": "^6.0.9",
    "@semantic-release/release-notes-generator": "^9.0.2",
    "@testing-library/dom": "^7.30.0",
    "@testing-library/jest-dom": "^5.11.9",
    "@testing-library/react": "^11.2.5",
    "@testing-library/user-event": "^12.8.3",
    "babel-jest": "^26.6.3",
    "commitlint-config-jira": "^1.5.1",
    "commitlint-plugin-jira-rules": "^1.5.1",
    "cypress": "^6.8.0",
    "eslint": "^7.22.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-cypress": "^2.11.2",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jest-dom": "^3.6.5",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.22.0",
    "eslint-plugin-react-hooks": "^4.2.0",
    "eslint-plugin-testing-library": "^3.10.1",
    "husky": "^5.1.3",
    "is-ci": "^3.0.0",
    "jest": "^26.6.3",
    "jest-next-dynamic": "^1.0.1",
    "lint-staged": "^10.5.4",
    "msw": "^0.27.1",
    "npm-merge-driver": "^2.3.6",
    "nyc": "^15.1.0",
    "prettier": "^2.2.1",
    "semantic-release": "^17.4.2",
    "start-server-and-test": "^1.12.1",
    "testcafe": "^1.12.0",
    "testcafe-browser-provider-browserstack": "^1.13.2-alpha.1"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "prettier": "blueprint/config/prettier",
  "eslintConfig": {
    "extends": [
      "./.blueprint/config/eslint"
    ]
  },
  "commitlint": {
    "extends": [
      "./.blueprint/config/commitlint"
    ]
  },
  "release": {
    "extends": "./.blueprint/config/release"
  },
  "husky": {
    "hooks": {
      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
      "pre-commit": "blueprint-scripts lint-staged"
    }
  }
}

```

**Comment by tommyboylab** [comment]
Same issue from my end, no postCSS config or use in my build (just Next's default config) and failing to build.

```
HookWebpackError: PostCSS plugin postcss-discard-comments requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
```

**Comment by yossisp** [comment]
I'm also having the issue. I'm using `"next": "^10.0.0",`. This is the error:
```
(node:32) UnhandledPromiseRejectionWarning: Error: PostCSS plugin postcss-discard-comments requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
    at Processor.normalize (/src/node_modules/postcss/lib/processor.js:167:15)
    at new Processor (/src/node_modules/postcss/lib/processor.js:56:25)
    at postcss (/src/node_modules/postcss/lib/postcss.js:55:10)
    at module.exports (/src/node_modules/cssnano-simple/src/index.js:54:10)
    at minifyCss (/src/node_modules/next/dist/build/webpack/plugins/font-stylesheet-gathering-plugin.js:1:554)
    at /src/node_modules/next/dist/build/webpack/plugins/font-stylesheet-gathering-plugin.js:25:432
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by timneutkens** [comment]
Instead of commenting on this issue with stacktraces please provide a full reproduction so that it can be investigated.

**Comment by mephysto** [comment]
Not sure if this is exactly your issue as well. But just adding my findings in here, since I had this same error message.

I am running on `"next": "^10.2.0",` so it should be leveraging the new Webfont Optimization feature. However, I am using Typekit as my  font library (instead of the more common Google Fonts). My `Meta.js` file:
```
import Head from "next/head";

const Meta = () => (
  <Head>
   ...
    
    <link rel="preload" href="https://use.typekit.net/abc123.css" as="style" />
    <link rel="stylesheet" href="https://use.typekit.net/abcd123.css" />
.
  </Head>
);

export default Meta;

```

My project started building when I removed my Typekit things.

Hope this helps in any way

**Comment by amosbastian** [comment]
I also encountered the same problem when including

```jsx
<link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700" rel="stylesheet" />
```

in my head component. I've tried reproducing it to no avail, but at least adding PostCSS 8 as a dev dependency solves it 😕

**Comment by mattiasw** [comment]
The only way we managed to solve this was to regenerate our `package-lock.json`.

**Comment by MantasMikal** [comment]
Also experiencing this issue.
Downgrading to `"next": "10.0.0"` fixed it for me

**Comment by timneutkens** [comment]
I'll repeat what I shared earlier:
Instead of commenting on this issue with stacktraces please provide a full reproduction so that it can be investigated.

**Comment by MantasMikal** [comment]
@timneutkens Here is the repo https://github.com/MantasMikal/next-postcss-issue
It seems that adding `storybook` and `sass` causes the issue. 
Thanks for your help

**Comment by timneutkens** [comment]
Thanks for the clear reproduction @MantasMikal, this allowed me to investigate and fix the issue. Opened a PR here: https://github.com/vercel/next.js/pull/26952

**Comment by haotangio** [comment]
Still have this problem in `"next": "^11.0.1"` while running `next build`. Anyone encounter the same issue?

**Comment by haotangio** [comment]
@timneutkens thanks for your response. I noticed that the `next build` is successful when I use Next 11.0.1 with SASS. It just fails when I upgrade from Next 10 to Next 11 - likely related to other libs in my `package.json`. So in the meantime I just force install Postcss 8 to fix the issue `npm i --save-dev postcss@8.2.13`.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
